src/Tribe/Views/V2/Views/Traits/HTML_Cache.php

```diff
@@ -0,0 +1,192 @@
+		 */
+		$cache_expiration = apply_filters( 'tribe_events_views_v2_cache_html_expiration', DAY_IN_SECONDS, $this->get_context(), $this );
+
+		$cache_key = $this->get_cache_html_key();
+
+		if ( ! $this->should_cache_html() ) {
+			return;
+		}
+
+		tribe( 'cache' )->set_transient( $cache_key, $html, $cache_expiration, $this->cache_html_triggers() );
+	}
+
+	/**
+	 * Fetch the HTML cache triggers.
+	 *
+	 * @since 5.0.0
+	 *
+	 * @return array
+	 */
+	protected function cache_html_triggers() {
+		$triggers = [
+			Cache_Listener::TRIGGER_SAVE_POST,
+			Cache_Listener::TRIGGER_UPDATED_OPTION,
+			Cache_Listener::TRIGGER_GENERATE_REWRITE_RULES,
+		];
+
+		return $triggers;
+	}
+
+	/**
+	 * Determine if HTML of the current view needs to be cached.
+	 *
+	 * @since 5.0.0
+	 *
+	 * @return bool
+	 */
+	public function should_cache_html() {
+		$context = $this->get_context();
+
+		$cached_views = [
+			'month',
+			'week',
+		];
+
+		$should_cache = in_array( $this->get_slug(), $cached_views ) && $this->should_enable_html_cache( $context );
+
+		/**
+		 * Should the v2 view HTML be cached?
+		 *
+		 * @since 5.0.0
+		 *
+		 * @param bool    $should_cache_html Should the current view have its HTML cached?
+		 * @param Context $context           The View current context.
+		 * @param View    $this              The current View instance.
+		 */
+		return (bool) apply_filters( 'tribe_events_views_v2_should_cache_html', $should_cache, $context, $this );
+	}
+
+	/**
+	 * Determine if HTML of the current view needs to be cached.
+	 *
+	 * @since 5.0.0
+	 *
+	 * @return bool
+	 */
+	public function get_cache_html_key() {
+		$context = $this->get_context();
+		$args    = $context->to_array();
+
+		unset( $args['now'] );
+
+		$cache_key = 'tribe_views_v2_cache_' . substr( sha1( json_encode( $args ) ), 0, 12 ) . ':';
+
+		/**
+		 * Filter the cached html key for v2 event views
+		 *
+		 * @since 5.0.0
+		 *
+		 * @param string  $cache_html_key Cache HTML key.
+		 * @param Context $context        The View current context.
+		 * @param View    $this           The current View instance.
+		 */
+		return apply_filters( 'tribe_events_views_v2_cache_html_key', $cache_key, $context, $this );
+	}
+
+	/**
+	 * Indicates if HTML cache should be enabled or not.
+	 *
+	 * If the HTML cache setting itself is not enabled (or not set) then this
+	 * method will always return false.
+	 *
+	 * In other cases, the default rules are to cache everything in the 2 months past
+	 * to 12 months in the future range. This policy can be refined or replaced via
+	 * the 'tribe_events_enable_month_view_cache' filter hook.
+	 *
+	 * @since 5.0.0
+	 *
+	 * @return bool
+	 */
+	protected function should_enable_html_cache( $context ) {
+		$event_date = $context->get( 'event_date' );
+
+		// Respect the month view caching setting
+		if ( ! tribe_get_option( 'enable_month_view_cache', true ) ) {
+			return false;
+		}
+
+		// Default to always caching the current month
+		if ( ! $event_date ) {
```

Comment:
:no_entry_sign: Inline comments must end in full-stops, exclamation marks, or question marks

<sub>hash: 7564cccfbf9e87e13049bdc0f2f17234</sub>